### DISCUSSION TITLE
How is the data/suggester file generated?

### DISCUSSION THREAD

**Discussion by qintian-qe** [oracle/opengrok#4305] [General]
Due to business requirements, only the index related files of the code need to be obtained. Opengrok was not deployed on Tomcat, but was directly used to generate the index file using Opengrok.jar. Later, it was found that there was no db file generated in the data/suggester directory. How are the files in the advisor directory generated?

**Comment by vladak** [comment]
The suggester data are created from within the webapp. What's your use case ?

**Reply by qintian-qe** [reply]
Firstly, generate index related files on different servers, and then consolidate the files generated on different servers onto a server deployed with web applications to achieve distributed implementation. This requires generating suggesters. You said that the advisor was generated by the webapp, which means that I directly put the index file into the relevant directory and restart Tomcat. Will the advisor be automatically generated? (Just tested, it is indeed automatically generated)

**Reply by qintian-qe** [reply]
This method of distributed index generation followed by merging is feasible after testing and verification, but it is unknown if there are any drawbacks.

**Reply by vladak** [reply]
> Firstly, generate index related files on different servers, and then consolidate the files generated on different servers onto a server deployed with web applications to achieve distributed implementation. This requires generating suggesters. You said that the advisor was generated by the webapp, which means that I directly put the index file into the relevant directory and restart Tomcat. Will the advisor be automatically generated? (Just tested, it is indeed automatically generated)

As long as the webapp knows about the project (it is present in the configuration it loads during deploy), it will generate the suggester data.

**Reply by vladak** [reply]
> This method of distributed index generation followed by merging is feasible after testing and verification, but it is unknown if there are any drawbacks.

In theory this should work, however I have never tried it myself. The biggest hurdle is project management (adding/deleting) and in general dealing with configuration merges.

**Reply by qintian-qe** [reply]
> > This method of distributed index generation followed by merging is feasible after testing and verification, but it is unknown if there are any drawbacks.
> 
> In theory this should work, however I have never tried it myself. The biggest hurdle is project management (adding/deleting) and in general dealing with configuration merges.

You're right, project management is indeed a bigger challenge that requires meticulous planning. I have tested this plan and it is feasible
